crates/runner/src/memory/mod.rs

```diff
@@ -42,7 +37,7 @@ pub struct Memory {
     /// allows methods with immutable `&self` receivers, like `get_data`, to
     /// modify the trace. This design choice separates the logical immutability
     /// of an operation from the implementation detail of tracing.
-    pub trace: RefCell<Vec<TraceEntry>>,
+    pub trace: RefCell<Vec<MemoryEntry>>,
```

Comment:
nit: 
Should an alias be used either in runner on in adapter?
seems a little off compared to pushing a traceEntry here